core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/PrintableField.java

```diff
@@ -0,0 +1,304 @@
+      return LimsUtils.formatDate(barcodable.getBarcodeDate());
+    }
+
+  },
+  DESCRIPTION {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.visit(new BarcodableVisitor<String>() {
+
+        @Override
+        public String visitKit(Kit kit) {
+          return kit.getKitDescriptor().getName();
+        }
+
+        @Override
+        public String visitLibrary(Library library) {
+          return library.getDescription();
+        }
+
+        @Override
+        public String visitPool(Pool pool) {
+          return pool.getDescription();
+        }
+
+        @Override
+        public String visitSample(Sample sample) {
+          return sample.getDescription();
+        }
+
+      });
+    }
+
+  },
+  EXTERNAL_NAME {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.visit(getParentSampleField(SampleIdentity.class, i -> i.getExternalName()));
+    }
+
+  },
+  GROUP_DESC {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.visit(new BarcodableVisitor<String>() {
+
+        @Override
+        public String visitLibraryAliquotDetailed(DetailedLibraryAliquot libraryAliquot) {
+          GroupIdentifiable groupParent = libraryAliquot.getEffectiveGroupIdEntity();
+          return groupParent == null ? null : groupParent.getGroupDescription();
+        }
+
+        @Override
+        public String visitLibraryDetailed(DetailedLibrary library) {
+          GroupIdentifiable groupParent = library.getEffectiveGroupIdEntity();
+          return groupParent == null ? null : groupParent.getGroupDescription();
+        }
+
+        @Override
+        public String visitSampleDetailed(DetailedSample sample) {
+          GroupIdentifiable groupParent = sample.getEffectiveGroupIdEntity();
+          return groupParent == null ? null : groupParent.getGroupDescription();
+        }
+
+      });
+    }
+  },
+  LABEL_TEXT {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.getLabelText();
+    }
+
+  },
+  LIBRARY_TYPE {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.visit(new BarcodableVisitor<String>() {
+
+        @Override
+        public String visitLibraryAliquotDetailed(DetailedLibraryAliquot libraryAliquot) {
+          return visitLibraryDetailed((DetailedLibrary) libraryAliquot.getLibrary());
+        }
+
+        @Override
+        public String visitLibraryDetailed(DetailedLibrary library) {
+          return library.getLibraryDesignCode().getCode();
+        }
+
+      });
+    }
+
+  },
+  NAME {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.getName();
+    }
+
+  },
+  SECONDARY_ID {
+
+    @Override
+    public String text(Barcodable barcodable) {
+      return barcodable.visit(getParentSampleField(SampleTissue.class, t -> t.getSecondaryIdentifier()));
+    }
+
+  },
+  SIZE {
```

Comment:
This seems intended to replace `Barcodable#getBarcodeSizeInfo`, which was previously returning concentration and volume information rather for samples, libraries, library aliquots, and pools. I imagine the `dnaSize` field from libraries and aliquots will be wanted on labels for the v2 naming scheme since it's not included in the alias, but the old layouts should still have concentration and volume

Reply:
Illinca is fine with the change and no one from GRP has responded, so forgiveness if it comes to it.